### PR TITLE
Updated OPtANNe PR

### PR DESCRIPTION
**Pull request by sourcesync**
@harsha-simhadri This PR should help us merge the OptANNe into main.  I've captured the essence of PRs 63 and 64 (originating from forked repos) here and if you accept/merge we can close 63 and 64.

### CODE REVIEW

File: algos.yaml [hunk 1]
```diff
@@ -537,6 +556,22 @@ deep-1B:
...
+          query-args: |
+              [{"search_window_size": 80}, 
+                {"search_window_size": 96},
+                {"search_window_size": 112},
+                {"search_window_size": 128},
+                {"search_window_size": 144},
+                {"search_window_size": 160}
+              ]
+
+>>>>>>> 7cd5540ffafefe956b751573f1314e723bba8f5c
```
Comment (harsha-simhadri) [inline algos.yaml:574]: Could you please fix this.

File: .python-version [hunk 2]
```diff
@@ -0,0 +1 @@
+3.8.12
```
Comment (harsha-simhadri) [inline .python-version:1]: Is this needed?

File: algos.yaml [hunk 3]
```diff
@@ -714,6 +753,27 @@ msspacev-1B:
                 {"nProbe":170, "efSearch":510, "K1":128, "blockSize":4096, "identifier": "bbann"},
                 {"nProbe":180, "efSearch":360, "K1":128, "blockSize":4096, "identifier": "bbann"}
               ]
+=======
```
Comment (harsha-simhadri) [inline algos.yaml:756]: Clean up markers from merge conflicts and verify the contents?

File: benchmark/datasets.py [hunk 4]
```diff
@@ -162,7 +162,7 @@ def read_ibin(filename, start_idx=0, chunk_size=None):
 
 
 def sanitize(x):
-    return numpy.ascontiguousarray(x, dtype='float32')
+    return numpy.ascontiguousarray(x)#, dtype='float32')
```
Comment (harsha-simhadri) [inline benchmark/datasets.py:165]: wouldn't this break main?

**Review by harsha-simhadri — CHANGES_REQUESTED** [review]

File: .python-version (hunk 2 above)
Comment (sourcesync) [reply .python-version]: I don't think so...

File: algos.yaml (hunk 3 above)
Comment (sourcesync) [reply algos.yaml:754]: yep

File: benchmark/datasets.py (hunk 4 above)
Comment (sourcesync) [reply benchmark/datasets.py]: yes good catch

**Comment by sourcesync** [comment]
@harsha-simhadri I see you asked @sdongaonkar about PR63.  I think it would be best to wait for his response on that PR before merging this one, since he may be in a position to test on the hardware.

**Pull request closed without merging**